src/ape/api/transactions.py

```diff
@@ -200,6 +203,56 @@ def decode_logs(self, abi: Union[EventABI, "ContractEvent"]) -> Iterator[Contrac
 
         yield from self.provider.network.ecosystem.decode_logs(abi, self.logs)
 
+    @property
+    def events(self):
+        contracts = {log["address"] for log in self.logs}
+        contract_types = {addr: self.chain_manager.contracts.get(addr) for addr in contracts}
+        decoded = []
+        for log in self.logs:
+            note = self.decode_ds_note(contract_types[log["address"]], log)
+            if note:
+                decoded.append(note)
+            else:
+                try:
+                    event_abi = next(
+                        abi
+                        for abi in contract_types[log["address"]].events
+                        if keccak(text=abi.selector) == log["topics"][0]
+                    )
+                    decoded.extend(self.provider.network.ecosystem.decode_logs(event_abi, [log]))
+                except StopIteration:
+                    decoded.append(log)
+
+        return decoded
+
+    def decode_ds_note(self, contract_type: ContractType, log: dict) -> Optional[ContractLog]:
+        selector, tail = log["topics"][0][:4], log["topics"][0][4:]
+        if sum(tail):
+            return None
+        try:
+            abi = next(
+                func
+                for func in contract_type.mutable_methods
+                if selector == keccak(text=func.selector)[:4]
+            )
+        except StopIteration:
+            return None
+
+        # in versions of ds-note the data field uses either (uint256,bytes) or (bytes) encoding
+        # instead of guessing, assume the payload starts right after the selector
+        data = decode_hex(log["data"])
+        input_types = [i.canonical_type for i in abi.inputs]
+        values = decode_abi(input_types, data[data.index(selector) + 4 :])  # noqa: E203
+
+        return ContractLog(  # type: ignore
```

Comment:
I think `ContractLog` misses the `address` field, it would be useful for dealing with logs emitted from multiple contracts